src/libmongoc/tests/test-mongoc-counters.c

```diff
@@ -26,31 +26,36 @@
  * those were superseded by write commands in 2.6. */
 #ifdef MONGOC_ENABLE_SHM_COUNTERS
 /* define a count function for each counter. */
-#define COUNTER(id, category, name, description)                \
-   uint32_t prev_##id;                                          \
-   uint32_t count_##id (void)                                   \
-   {                                                            \
-      uint32_t _sum = 0;                                        \
-      uint32_t _i;                                              \
-      for (_i = 0; _i < _mongoc_get_cpu_count (); _i++) {       \
-         _sum += __mongoc_counter_##id.cpus[_i]                 \
-                    .slots[COUNTER_##id % SLOTS_PER_CACHELINE]; \
-      }                                                         \
-      return _sum;                                              \
+#define COUNTER(id, category, name, description)                               \
+   int32_t prev_##id;                                                          \
+   int32_t count_##id (void)                                                   \
+   {                                                                           \
+      int32_t _sum = 0;                                                        \
+      int32_t _i;                                                              \
+      for (_i = 0; _i < _mongoc_get_cpu_count (); _i++) {                      \
```

Comment:
> warning: comparison of integers of different signs: 'int32_t' (aka 'int') and 'unsigned int' [-Wsign-compare]

I believe the `_i` counter should remain `uint32_t`.

Reply:
Good find. I don't know why I changed that.